src/pinnable_mapped_file.cpp

```diff
@@ -134,7 +134,8 @@ pinnable_mapped_file::pinnable_mapped_file(const std::filesystem::path& dir, boo
             std::filesystem::resize_file(_data_file_path, shared_file_size);
          }
          else if(shared_file_size < existing_file_size) {
-             std::cerr << "CHAINBASE: \"" << _database_name << "\" requested size of " << shared_file_size << " is less than "
+            _database_size = existing_file_size;
+            std::cerr << "CHAINBASE: \"" << _database_name << "\" requested size of " << shared_file_size << " is less than "
```

Comment:
I think this code would be clearer if the constructor used `_database_size` instead of `shared_file_size`.

Reply:
I don't disagree that it is not very clear. This is for a 5.0 issue, so I'd rather make a small change.

Also we have this (maybe unintuitive) behavior that we allow the user to request a smaller size, but if the database already exist and is larger we just use the existing size without shrinking it (which would be difficult).